doc-changelog/parse_pr_title.py

```diff
@@ -0,0 +1,345 @@
+    clean_title = clean_title.replace("`", "\\`").replace('"', '\\"')
+
+    # Save the clean pull request title as the CLEAN_TITLE environment variable
+    save_env_variable("CLEAN_TITLE", clean_title)
+
+
+def add_towncrier_config(org_name: str, repo_name: str, default_config: bool):
+    """Append the missing towncrier information to the pyproject.toml file.
+
+    Parameters
+    ----------
+    org_name: str
+        The name of the organization.
+    repo_name: str
+        The name of the repository.
+    default_config: bool
+        Whether or not to use the default towncrier configuration for the pyproject.toml file.
+    """
+    # Load pyproject.toml file
+    with open("pyproject.toml", "a+") as file:
+        config = toml.load("pyproject.toml")
+        tool = config.get("tool", "DNE")
+        towncrier = tool.get("towncrier", "DNE")
+
+        # List containing changelog sections under each release
+        changelog_sections = [
+            "added",
+            "dependencies",
+            "documentation",
+            "fixed",
+            "maintenance",
+            "miscellaneous",
+            "test",
+        ]
+
+        # Dictionary containing [tool.towncrier] keys and values
+        towncrier_config_sections = {
+            "directory": '"doc/changelog.d"',
+            "template": '"doc/changelog.d/changelog_template.jinja"',
+            "filename": {"web": '"doc/source/changelog.rst"', "repo": '"CHANGELOG.md"'},
+            "start_string": {
+                "web": '".. towncrier release notes start\\n"',
+                "repo": '"<!-- towncrier release notes start -->\\n"',
+            },
+            "title_format": {
+                "web": f'"`{{version}} <https://github.com/{org_name}/{repo_name}/releases/tag/v{{version}}>`_ - {{project_date}}"',
+                "repo": f'"## [{{version}}](https://github.com/{org_name}/{repo_name}/releases/tag/v{{version}}) - {{project_date}}"',
+            },
+            "issue_format": {
+                "web": f'"`#{{issue}} <https://github.com/{org_name}/{repo_name}/pull/{{issue}}>`_"',
+                "repo": f'"[#{{issue}}](https://github.com/{org_name}/{repo_name}/pull/{{issue}})"',
+            },
+        }
+
+        # Get the package name from [tool.flit.module]
+        flit = tool.get("flit", "DNE")
+        module = name = ""
+        if flit != "DNE":
+            module = flit.get("module", "DNE")
+            if module != ("DNE" or ""):
+                name = module.get("name", "DNE")
+                # If [tool.flit.module] name exists, create the package string
+                if name != ("DNE" and ""):
+                    towncrier_config_sections["package"] = f'"{name}"'
+
+        if default_config:
+            # If there is no towncrier configuration information or if [[tool.towncrier.type]]
+            # is the only towncrier information in the pyproject.toml file
+            if towncrier == "DNE" or len(towncrier) == 1:
```

Comment:
I have to check this because there can't be two [tool.towncrier] sections. If for some reason someone only has the [[tool.towncrier.type]] sections in their pyproject.toml, then len(towncrier) is 1